execution/gethexec/sequencer.go

```diff
@@ -1428,19 +1484,17 @@ func (s *Sequencer) Start(ctxIn context.Context) error {
 		})
 	}
 
-	s.CallIteratively(func(ctx context.Context) time.Duration {
-		nextBlock := time.Now().Add(s.config().MaxBlockSpeed)
-		if s.createBlock(ctx) {
-			// Note: this may return a negative duration, but timers are fine with that (they treat negative durations as 0).
-			return time.Until(nextBlock)
-		}
-		// If we didn't make a block, try again immediately.
-		return 0
-	})
-
 	return nil
 }
 
+func (s *Sequencer) StartSequencing(ctx context.Context) (*execution.SequencedMsg, time.Duration) {
+	sequencedMsg, waitUntilSequencingNextBlock := s.createBlock(ctx)
+	if waitUntilSequencingNextBlock {
+		return sequencedMsg, s.config().MaxBlockSpeed
```

Comment:
For now, to avoid a "config breaking change", I kept using this MaxBlockSpeed config provided in Execution instead of moving it to Consensus's config.